rules/CodeQuality/Rector/Class_/RemoveReadonlyPropertyVisibilityOnReadonlyClassRector.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Class_;
+use Rector\Php81\NodeManipulator\AttributeGroupNewLiner;
+use Rector\Privatization\NodeManipulator\VisibilityManipulator;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\MethodName;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodeQuality\Rector\Class_\RemoveReadonlyPropertyVisibilityOnReadonlyClassRector\RemoveReadonlyPropertyVisibilityOnReadonlyClassRectorTest
+ */
+final class RemoveReadonlyPropertyVisibilityOnReadonlyClassRector extends AbstractRector
+{
+    public function __construct(
+        private readonly VisibilityManipulator $visibilityManipulator,
+        private readonly AttributeGroupNewLiner $attributeGroupNewLiner
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove readonly property visibility on readonly class', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final readonly class SomeClass
+{
+    public function __construct(
+        private readonly string $name
+    ) {
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+final readonly class SomeClass
+{
+    public function __construct(
+        private string $name
+    ) {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node->isReadonly()) {
+            return null;
+        }
+
+        $constructClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+
+        if ($constructClassMethod instanceof ClassMethod) {
+            foreach ($constructClassMethod->getParams() as $param) {
+                $this->visibilityManipulator->removeReadonly($param);
```

Comment:
add `$param->isReadOnly()` check before, then add mark $hasChanged in the end before to return null if $hasChanged is false, same with Property